tests/common/range_index_space_id.h

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2017-2022 Codeplay Software LTD. All Rights Reserved.
+//  Copyright (c) 2022-2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#ifndef SYCL_CTS_COMPILING_WITH_HIPSYCL
+#include "../common/type_coverage.h"
+#endif
+
+namespace range_index_space_id {
+
+enum class check_codes : size_t {
+  dimensions_type = 0,
+  dimensions_value = 1,
+  code_count = 2,
+};
+
+constexpr size_t check_count = to_integral(check_codes::code_count);
+
+static const std::array<std::string, check_count> error_strings{
+    "Wrong type of member T<Dim>::dimensions",
+    "Wrong value of member T<Dim>::dimensions",
+};
+
+template <check_codes Code, typename ResultArray>
+void set_success_operation(ResultArray& result) {
+  int index = to_integral(Code);
+  result[index] = true;
+}
+
+inline std::string get_error_string(int code, const std::string& type_name) {
+  return error_strings[code] + " for type T '" + type_name + "'";
+}
+
+template <typename T, int Dim, typename ResultArray>
+void check_members(ResultArray& result) {
+  if (std::is_same_v<const int, decltype(T::dimensions)>) {
+    set_success_operation<check_codes::dimensions_type>(result);
+  }
+  if (Dim == T::dimensions) {
```

Comment:
```suggestion
  if constexpr (Dim == T::dimensions) {
```

Reply:
I prefer constexpr too, but in this case we'll get CI failure at compile time instead of fails at runtime. I thought runtime fails would be better. What do you think?

Reply:
Hmm, that's generally a bit of a philosophical question of whether to do compile time checks at compile time (we certainly don't everywhere in the CTS), but in this case it would have to compile regardless of `constexpr`, because `set_success_operation` will still need to be run at runtime for the test to pass. It's a minor change anyway, I don't mind much either way.